VM Flavor (#373)
openshift/openshift-origin-design: designs/virtualization/4.5/Scheduling/Flavor/Flavor.md

---
parent: virtualization
version: 4.5
---

# Flavor

Displaying the VM flavor (CPU and Memory) in the Scheduling section.

![vm page flavor medium](img/Flavor-0-0.jpg)
VM flavor is set on the creation process.
If the flavor is set to any of the sizes which is not custom, that flavor will be locked to changes.
Edit (pencil) button will be removed in this case.

![vm page flavor custom](img/Flavor-2-0.jpg)

> If the Flavor was set to Custom during the creation process, the flavor will be editable.

Author: I'm afraid so... @jelkosz correct me if I'm wrong:
In order for VM to be compatible with Kubevirt, it needs a common template (or flavor).
Once set, it cannot be modified. As I understand it, it is attached to the VM and not the VMI so turning the VM off will not help.
Furthermore, selecting Custom is a type of common template...
I think this scenario does not make sense for the user but it is a limitation we must work with.
We intend to add a note in the creation process that by selecting flavor size the VM flavor cannot be modified later.

> ![vm page flavor modal](img/Flavor-1-0.jpg)

Reviewer: spelling: Switch

> Opening the Edit Flavor modal reviels that the flavor is set and locked to Custom.

Reviewer: ```suggestion
Opening the Edit Flavor modal reveals that the flavor is set and locked to Custom.
```

> However, the values of the CPU is editabke in dedicated fields below.

Reviewer: ```suggestion
However, the values of the CPU is editable in dedicated fields below.
```

![vm page flavor modal](img/Flavor-1-1.jpg)

> By clicking 'Switch to full topology', the UI will toggle dispay full topoloy options.

Reviewer: ```suggestion
By clicking 'Switch to full topology', the UI will toggle to display full topology options.
```

> At the bottom of the "full topology" field, there will be a 'CPUs in total' counter.

Reviewer: I wonder if this should be more prominent.  Would there be an empty state/loading state for this?

Reviewer 2: Maybe you keep CPU where it is at all times and when the view the full topology you disable to actual editing of CPU and it just updates based on any topology edits? At that point, it would just be an "Expand to show full CPU topology" :)
